### ISSUE TITLE
RootRaisedCosine impulse response seem incorrect

### ISSUE THREAD

**Issue by dan-1d** [veeresht/CommPy#59]
When comparing a RRC impulse response generated by CommPy to those of other reference implementations, the results seem incorrect -- e.g., not even symmetric.
I don't have screenshots at the moment, since I changed implementations, but may try to show evidence here. 

Just a warning to those using it for pulse shaping.. it's not perfect.

**Comment by thomasfillon** [comment]
Indeed, all the filters functions return non-symmetric filters.

The time indexes are defined as:
```
T_delta = 1/float(Fs)
time_idx = ((np.arange(N)-N/2))*T_delta
```

As an example we have:
- for N = 5, T=_delta = 1, time_idx = [-2.5, -1.5, -0.5,  0.5,  1.5]
- for N = 6, T=_delta = 1, time_idx = [-3., -2., -1.,  0.,  1.,  2.]

Shouldn't the time indexes be symmetric like the following ?:
- for N=5,  time_idx = [-2., -1, 0.  1.,  2.]
- for N=6,  time_idx = [-2.5, -1.5, -0.5,  0.5,  1.5 ,  2.5]

which correspond to:
`time_idx = (np.arange(N)-(N-1)/2) * T_delta`

**Comment by BastienTr** [comment]
Hi @thomasfillon,

This could make sense. As I do not know enough about the module to provide a conclusive opinion, could you make the change and compare the result with a reference? That way, we would be assured that the issue is properly resolved.

**Comment by thomasfillon** [comment]
Hi @BastienTr ,

OK, I will send a pull request with the proposed code modification with some tests.
@dan-1d What  reference implementations were you comparing with ?

**Comment by dan-1d** [comment]
@thomasfillon   Good question! I think it was a MATLAB implementation in the "comm" toolbox, but it could have been in-house code. We ended up writing our own in Python, and I haven't gotten back to this since... I can perhaps compare the fix at some point.

**Comment by mobilinkd** [comment]
If you happen to choose a an even number of taps (80 in this case), no error is reported, but this is what you get:

```python
pos, taps = rrcosfilter(int(samples_per_symbol * 8), 0.5, 1.0/symbol_rate, sample_rate)
```
![bad_rrc](https://user-images.githubusercontent.com/4713996/98163510-89319880-1ea8-11eb-9205-31447b1ac4ca.png)

**Comment by Fangrf** [comment]
Yes I use it for pulse shaping, it's not correct. When the nums_tap is odd, t = 0 would not satisfy.

**Comment by BastienTr** [comment]
As I previously said, the code was written by @veeresht before I joined the project. It is out of my knowledge scope.

If some of you have a correct code and a reference to cross check, I'd happily merge the PR and update the module. Unfortunately I can't do it myself because I don't know enough on these topics.

**Comment by Fangrf** [comment]
Sorry to bother you! At the beginning, I just didn't see this issue. When I was plotting the constellation diagram after pulse shaping, I encountered the strange phenomenon of the constellation diagram being very dispersed. Finally, by comparing with MATLAB-related functions, I found that it was a problem with the RRC filter. In the end, I wrote an RRC filter myself to replace it, which solved the problem. 

Since the original author is no longer updating it, please just ignore it. I originally just wanted to remind future users, but having too many still open issues might cause people to overlook it.

**Comment by Fangrf** [comment]
By plotting the time-domain diagram of the RRC filter in MATLAB, it can be seen that when the number of taps is odd, the very middle sample corresponds to time 0. In the implementation of commpy, “time_idx = ((np.arange(N)-N/2))*T_delta”, which results in complete symmetry when the number of taps is odd, and the middle sample does not align with time 0. In particular, it should be noted that RRC filters are generally assumed to have an odd number of taps. In MATLAB, for rcosdesign(rolloff, span, sps), span*sps must be even, and the number of taps will be equal to span*sps + 1.

**Comment by BastienTr** [comment]
Would you like to make a PR with your adapted RRC function? Matlab looks a good enough reference to me. 

Even if the original author is no more active, I'm still ok to maintain the code and review PR. I just can't resolve the issues and make new code on my own
